modules/drivers/mongo/src/metabase/driver/mongo.clj

```diff
@@ -204,6 +205,24 @@
     {:tables (set (for [collection (disj (mdb/get-collection-names conn) "system.indexes")]
                     {:schema nil, :name collection}))}))
 
+(defmethod driver/describe-table-indexes :mongo
+  [_ database table]
+  (with-mongo-connection [^com.mongodb.DB conn database]
+    (->> (mcoll/indexes-on conn (:name table))
+         (map :key)
+         ;; when the key is not a persistent array map, then the ordered of the key can't be determined
+         ;; so we ignore those.
+         ;; Practically this will be only the case for when an index has more than 8 keys because a map
+         ;; with less than 8 keys are PersistentArrayMap, it's a PersistentHashedMap if it has more than 8 keys.
+         (filter #(instance? clojure.lang.PersistentArrayMap %))
```

Comment:
It would be great to remove this edge case. Instead of calling `mcoll/indexes-on`, perhaps you could implement your own function that gets the indexes in a data structure that preserves ordering regardless of the size of the compound index?

Reply:
I thought about that but it's a very low-level stuff in mongo, basically, we need to redefine a protocol for [ConvertToDBObject](https://github.com/michaelklishin/monger/blob/84170f7c51973e3d789736cdeb3cd8ad013071c3/src/clojure/monger/conversion.clj#L52).

We shouldn't do this as it'll affect the performance when reading large objects.

Also, I think a compound key with > 8 keys is quite rare.

Reply:
I don't think you need to redefine the monger protocol, you can just use the Java API directly in this one specific use case. Like start with `(.getIndexInfo (.getCollection db (name coll)))`.

> Also, I think a compound key with > 8 keys is quite rare.

Yes, it's probably not worth fixing it for customers, but I think it's worth fixing just because this edge case complicates the code and I have to read this huge comment to understand what's it's doing.

Reply:
ha, I didn't realize you can access the dbobject like a normal object :). Good to know!
Implemented here https://github.com/metabase/metabase/pull/36624/files/31b0a616a7861fd1aceec4a35584bfc4cc527918..f683341abfd69eddd31d4ed1f72f74709770a8f9

Thanks Cal!